src/components/switch/switch.ts

```diff
@@ -0,0 +1,138 @@
+import { html, LitElement } from 'lit';
+import { property, query, state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { live } from 'lit/directives/live.js';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import { Constructor } from '../common/mixins/constructor.js';
+import { styles } from './switch.material.css';
+import { watch } from '../common/decorators/watch.js';
+
+let nextId = 0;
+
+export interface IgcSwitchEventMap {
+  igcChange: CustomEvent<void>;
+  igcFocus: CustomEvent<void>;
+  igcBlur: CustomEvent<void>;
+}
+
+// @customElement('igc-switch')
+export class IgcSwitchComponent extends EventEmitterMixin<
+  IgcSwitchEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  static styles = styles;
+
+  private inputId = `switch-${nextId++}`;
+  private labelId = `switch-label-${this.inputId}`;
+
+  @query('input[type="checkbox"]')
```

Comment:
```suggestion
  @query('input[type="checkbox"]', true)
```

Since the input is always present in the shadow DOM better cache it.